game/ui/components/equipscreen.cpp

```diff
@@ -60,10 +61,18 @@ void EquipmentPaperDoll::setNonHighlightColour(const Colour &colour)
 
 void EquipmentPaperDoll::update()
 {
+	static unsigned long lasttime = 0;
+	unsigned long currenttime, deltatime;
+
 	// Only draw highlight if one of the colours isn't 100% transparent
 	if (slotHighlightColours[0].a != 0 || slotHighlightColours[1].a != 0)
 	{
-		this->slotHighlightCounter += slotHighlightIncrement;
+		currenttime = SDL_GetTicks();
```

Comment:
Never use SDL outside the framework. Do you really need the global time here? Can't you just scale the local counter time differently if you want to slow it down?

If you think you /really/ need to have this based on real time (and not based on the number of times the screen is updated, which does depend on frame rate) add a function to the Framework instead of calling sdl directly.

Reply:
Yes, I need real time. Time between frames will be perfect. But I don't know where to get it. Local counter won't work because mousecursor movement changes framerate 4 times and flashing becomes 4 times slower. Can you explain me where can I get milliseconds last between frames?

Reply:
Well I thinks it's a good idea to leave it a local var for now. Everything uses it (pulsating minimap icons, unit waypoint icons, healing icon etc.). Yes it may be imperfect but it will be consistent across the game. 

You can plan a task in Trello to rework this system everywhere in the game, so that we don't have a myriad of local vars to track pulsation, maybe it's worth doing eventually.